src/vivarium_gates_mncnh/components/screening.py

```diff
@@ -90,29 +94,24 @@ def on_time_step_cleanup(self, event: Event) -> None:
             .astype("object")
             .fillna("not_anemic")
         )
+        self.population_view.update(later_anc_pop)
 
-        self.population_view.update(anc_pop)
-
-        # hemoglobin screening
+        # determine who gets hemoglobin screening
         if INTERVENTION_SCENARIOS[self.scenario].hemoglobin_screening_coverage == "baseline":
-            hemoglobin_screening = self.randomness.choice(
-                index=anc_pop.index,
-                choices=[True, False],
-                p=[
-                    self.hemoglobin_screening_coverage,
-                    1 - self.hemoglobin_screening_coverage,
-                ],
-                additional_key="hemoglobin_screening_coverage",
+            screen_for_hemoglobin = (
+                propensity[later_anc_pop.index] < self.hemoglobin_screening_coverage
             )
-            hemoglobin_screening.name = COLUMNS.HEMOGLOBIN_SCREENING_COVERAGE
+            screen_for_hemoglobin.name = COLUMNS.HEMOGLOBIN_SCREENING_COVERAGE
         else:
-            hemoglobin_screening = pd.DataFrame(
-                {COLUMNS.HEMOGLOBIN_SCREENING_COVERAGE: True}, index=anc_pop.index
-            )
+            # screen all eligible simulants
+            screen_for_hemoglobin = pd.Series(False, index=pop.index)
+            screen_for_hemoglobin.loc[later_anc_pop.index] = True
+            screen_for_hemoglobin.name = COLUMNS.HEMOGLOBIN_SCREENING_COVERAGE
 
-        self.population_view.update(hemoglobin_screening)
+        self.population_view.update(screen_for_hemoglobin)
```

Comment:
Why are we updating this column in the state table and then pulling the full state table again? Why not just update `pop` directly, since we send `pop` to `update()` at the end of this method?